pkg/sources/git/git.go

```diff
@@ -637,15 +638,23 @@ func normalizeConfig(scanOptions *ScanOptions, repo *git.Repository) (err error)
 }
 
 // GenerateLink crafts a link to the specific file from a commit. This works in most major git providers (Github/Gitlab)
-func GenerateLink(repo, commit, file string) string {
+func GenerateLink(repo, commit, file string, line int64) string {
 	// bitbucket links are commits not commit...
 	if strings.Contains(repo, "bitbucket.org/") {
 		return repo[:len(repo)-4] + "/commits/" + commit
 	}
-	link := repo[:len(repo)-4] + "/blob/" + commit + "/" + file
-
+	var link string
 	if file == "" {
 		link = repo[:len(repo)-4] + "/commit/" + commit
+	} else {
+		link = repo[:len(repo)-4] + "/blob/" + commit + "/" + file
```

Comment:
nit: I think if the line is 0 (not sure if this would happen, maybe if we are using the zero value), we would append `L#0` to the url. I don't think it matters too much since both github and gitlab appear to just show you the page, but figured i'd mention it. We could also check if line is 0 and return the link before doing the contains check and concatentation.

Reply:
I've updated the logic accordingly.

I'm also wondering whether it's worth passing in the type of source, rather than relying on the presence of "github" or "gitlab" in the URL. Right now only `github.go` and `gitlab.go` seem to call `git.GenerateLink`, so it should always generate the link.

Reply:
I think it seems okay for now given the points you mentioned. I also feel like we could also potentially refactor this at some point to remove the magic strings throughout the function. Including at the start where we do a check for if it's "bitbucket".